test/testffmpeg.c

```diff
@@ -137,18 +138,41 @@ static SDL_bool CreateWindowAndRenderer(Uint32 window_flags, const char *driver)
 
 #ifdef HAVE_EGL
     if (useEGL) {
-        const char *extensions = eglQueryString(eglGetCurrentDisplay(), EGL_EXTENSIONS);
-        if (SDL_strstr(extensions, "EGL_EXT_image_dma_buf_import") != NULL) {
-            has_EGL_EXT_image_dma_buf_import = SDL_TRUE;
+        const char *egl_extensions = eglQueryString(eglGetCurrentDisplay(), EGL_EXTENSIONS);
+        if (!egl_extensions) {
+            return SDL_FALSE;
         }
 
+        char *extensions = SDL_strdup(egl_extensions);
+        if (!extensions) {
+            return SDL_FALSE;
+        }
+
+        char *saveptr, *token;
+        token = SDL_strtok_r(extensions, " ", &saveptr);
+        if (!token) {
+            free(extensions);
+            return SDL_FALSE;
+        }
+        do {
+            SDL_Log("EGL extension '%s' is supported", token);
+            if (SDL_strcmp(token, "EGL_EXT_image_dma_buf_import") == 0) {
+                has_EGL_EXT_image_dma_buf_import = SDL_TRUE;
+            } else if (SDL_strcmp(token, "EGL_EXT_image_dma_buf_import_modifiers") == 0) {
+                has_EGL_EXT_image_dma_buf_import_modifiers = SDL_TRUE;
+            }
+        } while ((token = SDL_strtok_r(NULL, " ", &saveptr)) != NULL);
+
+        free(extensions);
+
         if (SDL_GL_ExtensionSupported("GL_OES_EGL_image")) {
             glEGLImageTargetTexture2DOESFunc = (PFNGLEGLIMAGETARGETTEXTURE2DOESPROC)eglGetProcAddress("glEGLImageTargetTexture2DOES");
         }
 
         glActiveTextureARBFunc = (PFNGLACTIVETEXTUREARBPROC)SDL_GL_GetProcAddress("glActiveTextureARB");
 
         if (has_EGL_EXT_image_dma_buf_import &&
+            has_EGL_EXT_image_dma_buf_import_modifiers &&
```

Comment:
It's possible for the implementation to support EGL_EXT_image_dma_buf_import  and not EGL_EXT_image_dma_buf_import_modifiers. This code should still run, just not add the modifiers, in this case.

Reply:
Thanks, made it conditional in 6ede48514061e9ca941cc4f02d761cc83c0ef3fb.